pkg/runtime/reconciler.go

```diff
@@ -292,12 +288,16 @@ func (r *resourceReconciler) ensureConditions(
 	exit := rlog.Trace("r.ensureConditions")
 	defer exit(err)
 
-	if syncedCond := ackcondition.Synced(res); syncedCond == nil {
-		rlog.Debug("resource missing ACK.ResourceSynced condition")
-		// only the resource manager will know whether the resource is in a
-		// stable sync state. Even if we got no error back from the
-		// create/update operations, we can only set this to Unknown.
-		condStatus := corev1.ConditionUnknown
+	// If the ACK.ResourceSynced condition is not set using the custom hooks,
+	// determine the Synced condition using "rm.IsSynced" method
+	if ackcondition.Synced(res) == nil {
```

Comment:
Do we want to check for `Unknown` as well?

Reply:
Before this line in code, only custom hooks set the `ACK.ResourceSynced` condition, and I am sure none of them set is as `Unknown`. Now the recommendation is to use `IsSynced` method rather than custom hooks, so I do not think we need to check for `Unknown` status.